parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -85,6 +87,10 @@
     this.rowGroups = reader.getRowGroups();
     this.shouldSkip = new boolean[rowGroups.size()];
 
+    // Fetch all row groups starting positions to compute the row offsets of the filtered row groups
+    Map<Long, Long> offsetToStartPos = generateOffsetToStartPos();
```

Comment:
It just occurred to me (after merging this) that we may want to make this lazy, like we do in Avro. That way if the row positions are never used, we don't incur the cost of reading the footer another time.

Reply:
I used to think to apply `Caffeine` cache this. Let me think about this again and update this in follow up vectorization code path.

Reply:
@chenjunjiedada  yes we should make this lazy , do you have issue to track improvements to existing logic?

Reply:
@sudssf , Yes, I have a PR: https://github.com/apache/iceberg/pull/1356. It reads the file when the required schema contains position column.

```java
    if (expectedSchema.findField(MetadataColumns.ROW_POSITION.fieldId()) != null) {
      // Only read footer when needed
      offsetToStartPos = generateOffsetToStartPos();
    }
```